File: 62/Project_Euler_62.py
from itertools import permutations


def isCube(n):
    i = 1
    while(pow(i, 3) < n):
        i += 1
    
    return n == pow(i, 3)


k = 345
found = False
while(not found):
    cube = pow(k, 3)
    counter = 1
    
    perms = [''.join(p) for p in permutations(str(cube))]

    for i in range(1, len(perms)):
        if(perms[i][0] != 0):
            if isCube(int(perms[i])):
                counter += 1
    
    if(counter == 5):
        print(k)
        found = True

    print("Number of cubes: " + str(counter))

    k += 1


    

File: 62/Project_Euler_62.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
#include <string>

using ull = unsigned long long;

bool isCube(ull n)
{   
    ull temp {static_cast<ull>(std::cbrt(n))};
    return temp * temp * temp == n;
}

int main()
{
    bool found {false};
    ull k {1};
    
    while(!found)
    {
        ull cube{k * k * k};
        int counter {0};
        std::string s {std::to_string(cube)};
        do
        {   
            if(s[0] != 0 && isCube(std::stoull(s))) ++counter;
            
        } while(std::next_permutation(s.begin(), s.end()));

        if(counter == 5)
        {
            std::cout << "Solution: " << cube << " k: " << k << '\n';
            found = true;
        }

        //std::cout << counter << '\n';
        ++k;
    }

    return 0;
}